.github/workflows/lintrunner_ci.yml

```diff
@@ -0,0 +1,123 @@
+name: Lintrunner CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.event_name == 'workflow_dispatch' }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - name: Initialize git config for testing
+        run: |
+          git config --global user.email "fake@email.com"
+          git config --global user.name "mr fake"
+      - name: Install and initialize sapling on macos
+        if: matrix.os == 'macos-latest'
+        run: |
+          brew install sapling
+          sl config --user ui.username "mr fake <fake@email.com>"
+      - name: Run cargo test
+        uses: nick-fields/retry@v2.8.2
+        env:
+          TMPDIR: ${{ runner.temp }}
+        with:
+          timeout_minutes: 10
+          max_attempts: 10
+          retry_wait_seconds: 90
+          command: |
+            cd lintrunner
+            cargo test
+
+
+  linux:
+    runs-on: ubuntu-latest
+    needs: test
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+    - uses: messense/maturin-action@v1
+      with:
+        manylinux: auto
+        command: build
+        args: --release -o tools/lintrunner/target/dist
+        working-directory: tools/lintrunner
+    - name: Upload wheels
+      uses: actions/upload-artifact@v2
+      with:
+        name: wheels
+        path: tools/lintrunner/target/dist
+
+  windows:
+    runs-on: windows-latest
+    needs: test
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+    - uses: messense/maturin-action@v1
+      with:
+        command: build
+        args: --release -o dist
+        working-directory: lintrunner
+    - name: Upload wheels
+      uses: actions/upload-artifact@v2
+      with:
+        name: wheels
+        path: dist
+
+  macos:
+    runs-on: macos-latest
+    needs: test
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+    - uses: messense/maturin-action@v1
+      with:
+        command: build
+        args: --release -o dist --universal2
+        working-directory: lintrunner
+    - name: Upload wheels
+      uses: actions/upload-artifact@v2
+      with:
+        name: wheels
+        path: dist
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    if: "startsWith(github.ref, 'refs/tags/')"
```

Comment:
As we have both lintrunner and torchfix in the same repo, does this make sense to have more specific release tags for them?  `refs/tags/v*` works well if lintrunner stands alone.  A simple tweak would be to have something like `refs/tags/lintrunner/v*`, any thoughts? This doesn't need to be done in the same PR though IMO but we probably need it later